truffle/src/main/java/org/jruby/truffle/core/format/format/FormatCharacterNode.java

```diff
@@ -0,0 +1,85 @@
+package org.jruby.truffle.core.format.format;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.NodeChild;
+import com.oracle.truffle.api.dsl.NodeChildren;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import org.jruby.truffle.RubyContext;
+import org.jruby.truffle.core.format.FormatNode;
+import org.jruby.truffle.core.format.LiteralFormatNode;
+import org.jruby.truffle.core.format.convert.ToIntegerNode;
+import org.jruby.truffle.core.format.convert.ToIntegerNodeGen;
+import org.jruby.truffle.core.format.convert.ToStringNode;
+import org.jruby.truffle.core.format.convert.ToStringNodeGen;
+import org.jruby.truffle.core.format.exceptions.NoImplicitConversionException;
+import org.jruby.truffle.core.format.write.bytes.WriteByteNodeGen;
+import org.jruby.truffle.language.control.RaiseException;
+
+import java.nio.charset.StandardCharsets;
+
+@NodeChildren({
+    @NodeChild(value = "width", type = FormatNode.class),
+    @NodeChild(value = "value", type = FormatNode.class),
+})
+public abstract class FormatCharacterNode extends FormatNode {
+
+    private final boolean hasMinusFlag;
+
+    @Child private ToIntegerNode toIntegerNode;
+    @Child private ToStringNode toStringNode;
+
+    public FormatCharacterNode(RubyContext context, boolean hasMinusFlag) {
+        super(context);
+        this.hasMinusFlag = hasMinusFlag;
+    }
+
+//    @TruffleBoundary
```

Comment:
We try not to leave commented code like this in. At least it should be explained why it is currently commented.